tests/load/duckdb/test_motherduck_client.py

```diff
@@ -47,15 +52,49 @@ def test_motherduck_configuration() -> None:
     assert config.password == "tok"
 
 
-def test_motherduck_connect() -> None:
+def test_motherduck_connect(mocker: MockerFixture) -> None:
+    # set HOME env otherwise some internal components in ducdkb (HTTPS) do not initialize
+    os.environ["HOME"] = "/tmp"
+
+    connect_spy = mocker.spy(duckdb, "connect")
+    config = resolve_configuration(
+        MotherDuckClientConfiguration()._bind_dataset_name(dataset_name="test"),
+        sections=("destination", "motherduck"),
+    )
+    # connect
+    con = config.credentials.borrow_conn(read_only=False)
+    con.sql("SHOW DATABASES")
+    config.credentials.return_conn(con)
+
+    # check for the default user agent value
+    connect_spy.assert_called()
+    assert "config" in connect_spy.call_args.kwargs
+    assert connect_spy.call_args.kwargs["config"]["custom_user_agent"] == MOTHERDUCK_USER_AGENT
+
+
+def test_motherduck_connect_disable_user_agent_passing(mocker: MockerFixture) -> None:
```

Comment:
this test and the above one are almost the same, it would be better to parametrize them with the setting, it has two upsides
1. less code
2. you can really check that the the custom_user_agent is the only thing that changes between the test runs.